File: problems-extend-adapt/2201-2300/2274_longest-stretch-of-open-floors/solution.py
class Solution:
    def longestOpenStretch(self, bottom: int, top: int, blocked: List[int]) -> int:
        floors = sorted(blocked)
        best = max(floors[0] - bottom, top - floors[-1])
        for previous, floor in zip(floors, floors[1:]):
            best = max(best, floor - previous - 1)
        return best

File: problems-extend-adapt/2201-2300/2274_longest-stretch-of-open-floors/solution.cpp
class Solution {
  public:
    int longestOpenStretch(int bottom, int top, vector<int> &blocked) {
        sort(blocked.begin(), blocked.end());
        int best = max(blocked.front() - bottom, top - blocked.back());
        for (size_t i = 1; i < blocked.size(); i++) {
            best = max(best, blocked[i] - blocked[i - 1] - 1);
        }
        return best;
    }
};
